src/indexer/log_merge_policy.rs

```diff
@@ -54,40 +54,35 @@ impl LogMergePolicy {
 
 impl MergePolicy for LogMergePolicy {
     fn compute_merge_candidates(&self, segments: &[SegmentMeta]) -> Vec<MergeCandidate> {
-        let mut size_sorted_tuples = segments
+        let mut size_sorted_segments = segments
             .iter()
-            .map(SegmentMeta::num_docs)
-            .enumerate()
-            .filter(|(_, s)| s <= &(self.max_merge_size as u32))
-            .collect::<Vec<(usize, u32)>>();
+            .filter(|s| s.num_docs() <= (self.max_merge_size as u32))
+            .collect::<Vec<&SegmentMeta>>();
 
-        size_sorted_tuples.sort_by(|x, y| y.1.cmp(&(x.1)));
-
-        if size_sorted_tuples.len() <= 1 {
-            return Vec::new();
+        if size_sorted_segments.len() <= 1 {
+            return vec![];
         }
+        size_sorted_segments.sort_by_key(|seg| seg.num_docs());
 
-        let size_sorted_log_tuples: Vec<_> = size_sorted_tuples
+        let sorted_segments_with_log_size: Vec<_> = size_sorted_segments
             .into_iter()
-            .map(|(ind, num_docs)| (ind, f64::from(self.clip_min_size(num_docs)).log2()))
+            .map(|seg| (seg, f64::from(self.clip_min_size(seg.num_docs())).log2()))
             .collect();
 
-        if let Some(&(first_ind, first_score)) = size_sorted_log_tuples.first() {
-            let mut current_max_log_size = first_score;
-            let mut levels = vec![vec![first_ind]];
-            for &(ind, score) in (&size_sorted_log_tuples).iter().skip(1) {
-                if score < (current_max_log_size - self.level_log_size) {
-                    current_max_log_size = score;
+        if let Some(&(first_segment, log_size)) = sorted_segments_with_log_size.first() {
+            let mut current_max_log_size = log_size;
+            let mut levels = vec![vec![first_segment]];
+            for &(segment, segment_log_size) in (&sorted_segments_with_log_size).iter().skip(1) {
```

Comment:
isn't it the same without the parenthesis and the &?

Reply:
```
for &(segment, segment_log_size) in sorted_segments_with_log_size.iter().skip(1) {
```

Reply:
yes, I assume someone started with an iter on the ref &sorted_segments_with_log_size and added the skip later